src/coreclr/tools/aot/ILCompiler.ReadyToRun/JitInterface/CorInfoImpl.ReadyToRun.cs

```diff
@@ -921,13 +923,7 @@ private void getFunctionEntryPoint(CORINFO_METHOD_STRUCT_* ftn, ref CORINFO_CONS
 
         private bool canTailCall(CORINFO_METHOD_STRUCT_* callerHnd, CORINFO_METHOD_STRUCT_* declaredCalleeHnd, CORINFO_METHOD_STRUCT_* exactCalleeHnd, bool fIsTailPrefix)
         {
-            if (fIsTailPrefix)
-            {
-                // FUTURE: Delay load fixups for tailcalls
-                throw new RequiresRuntimeJitException(nameof(fIsTailPrefix));
-            }
-
-            return false;
+            return true;
```

Comment:
This cannot be just true. It must respect the same set of rules that are implemented in canTailCall in jitinterface.cpp. Notably if there is no tail prefix the following situations are forbidden from tail call optimization
- If the Caller method is NoInline 
- If the Caller method is the entrypoint to the application
- If the Callee method has the RequireSecObject bit set on the MethodDef

Reply:
Addressed this, PTAL.